### ISSUE TITLE
[BUG] a_star fails on 1.2

### ISSUE THREAD

**Issue by Katsutoshii** [JuliaGraphs/LightGraphs.jl#1258]
`a_star` throws the following error when updating a distance:
```
ERROR: LoadError: ArgumentError: PriorityQueue keys must be unique
Stacktrace:
 [1] enqueue!(::DataStructures.PriorityQueue{Integer,Float64,Base.Order.ForwardOrdering}, ::Pair{Integer,Float64}) at C:\Users\joshi\.julia\packages\DataStructures\SAI1X\src\priorityqueue.jl:236
 [2] enqueue! at C:\Users\joshi\.julia\packages\DataStructures\SAI1X\src\priorityqueue.jl:252 [inlined]
 [3] enqueue! at C:\Users\joshi\.julia\packages\DataStructures\SAI1X\src\priorityqueue.jl:251 [inlined]
 [4] a_star_impl!(::SimpleGraph{Int64}, ::Int64, ::DataStructures.PriorityQueue{Integer,Float64,Base.Order.ForwardOrdering}, ::Array{Bool,1}, ::Array{Float64,1}, ::Array{Float64,1}, ::Array{Int64,1}, ::Array{Float64,2}, ::getfield(LightGraphs, Symbol("##103#104")){Float64}) at C:\Users\joshi\.julia\dev\LightGraphs\src\shortestpaths\astar.jl:50
```

Reproducible by running the following with Julia 1.2, DataStructures 0.17.5:
```julia
using LightGraphs

function astarbug()
    g::SimpleGraph = complete_graph(4)
    w::Array{Float64, 2} = [
        1. 1. 1. 4.;
        1. 1. 1. 1.;
        1. 1. 1. 1.;
        4. 1. 1. 1.;
    ]
    res = a_star(g, 1, 4, w)
end

astarbug()

```

**Version information**

```
Julia Version 1.2.0
Commit c6da87ff4b (2019-08-20 00:03 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Core(TM) i7-7700K CPU @ 4.20GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.1 (ORCJIT, skylake)
Environment:
  JULIA_HOME = C:\Users\joshi\AppData\Local\Julia-1.2.0\bin
```

```
    Status `C:\Users\joshi\.julia\dev\MotionPlanning\Project.toml`
  [864edb3b] ↑ DataStructures v0.17.3 ⇒ v0.17.5
  [093fc24a]   LightGraphs v1.3.0
  [37e2e46d]   LinearAlgebra
    Status `C:\Users\joshi\.julia\dev\MotionPlanning\Manifest.toml`
  [864edb3b] ↑ DataStructures v0.17.3 ⇒ v0.17.5
```

**Comment by Katsutoshii** [comment]
Fixable by replacing `enqueue!(open_set, neighbor, priority)` with `open_set[neighbor] = priority` on line 50 of `astar.jl`

**Comment by sbromberger** [comment]
@Katsutoshii - thanks. Would you mind making a PR?

**Comment by sbromberger** [comment]
Ref: #1271

**Comment by sbromberger** [comment]
Thanks for the report. Fixed via #1271.
